File: framework_lib/src/main/java/com/base/BaseCommonActivity.java
package com.base;

import android.databinding.DataBindingUtil;
import android.databinding.ViewDataBinding;
import android.os.Bundle;
import com.navigation.BaseActivity;
import com.navigation.BaseViewModel;
import com.widgets.ToastCompat;
import org.jetbrains.annotations.Nullable;

/**
 * @author ysk
 * @class describe  {@link #}
 * @time 2019/4/23 下午4:50
 */
public abstract class BaseCommonActivity<B extends ViewDataBinding, VM extends BaseViewModel> extends BaseActivity {


    private B mBinding;
    private VM mViewModel;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        getIntentParams();
        super.onCreate(savedInstanceState);

        mBinding = DataBindingUtil.setContentView(this, getLayoutId());
        if (getVariableId() != 0) {
            mBinding.setVariable(getVariableId(), mViewModel);
            mBinding.executePendingBindings();
        }
        mViewModel.create();
        initGlobalParams();
    }

    private void initGlobalParams() {

    }


    private void getIntentParams() {

    }

    protected abstract int getLayoutId();

    protected abstract VM createViewModel();

    protected abstract int getVariableId();

    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (mViewModel != null) {
            mViewModel.onDestroy();
            mViewModel = null;
        }
    }

    public void showToast(String string) {
        ToastCompat.Companion.showToast(this, string);
    }

    public void showToast(int string) {
        ToastCompat.Companion.showToast(this, String.valueOf(string));
    }
}

File: framework_lib/src/main/java/com/base/BaseCommonActivity.kt
package com.base

import android.databinding.DataBindingUtil
import android.databinding.ViewDataBinding
import android.os.Bundle
import com.navigation.BaseActivity
import com.navigation.BaseViewModel

/**
 * @author ysk
 * @class describe  [.]
 * @time 2019/4/20 下午12:42
 */
abstract class BaseCommonActivity<B : ViewDataBinding, VM : BaseViewModel> : BaseActivity() {

    var binding: B? = null
        protected set
    var viewModel: VM? = null
        protected set

    protected abstract val layoutId: Int

    protected abstract val variableId: Int

    public override fun onCreate(savedInstanceState: Bundle?) {
        getIntentParams()
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, layoutId)
        if (variableId != 0) {
            binding?.setVariable(variableId, viewModel)
            binding?.executePendingBindings()
        }
        viewModel?.create()
        initGlobalParams()
    }


    protected abstract fun initGlobalParams()


    public override fun onDestroy() {
        super.onDestroy()
        viewModel?.onDestroy()
    }

    protected abstract fun createViewModel(): VM

    fun initStateBar() {

    }

    fun getIntentParams() {

    }


}
